### PR TITLE
V4 - theme refactoring, documentation and code improvment

### PR DESCRIPTION
**Pull request by JeremyDolle**

### CODE REVIEW

File: documentation/docs/04-Guides/01-Navigate.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+---
+
+All navigation within the app is accomplished using [React Navigation](https://reactnavigation.org/). 
+We adhere to the default navigation options without any overrides, custom navigators, or special configurations. 
+This approach allows us to stay aligned with the latest releases of React Navigation, ensuring that we can swiftly adopt 
+any new features and improvements.
+
+## Navigation structure
+
+All navigation-related configurations and navigators are neatly organized within the `src/navigation` folder. Here's a brief overview:
```
Comment (jgautrais) [inline documentation/docs/04-Guides/01-Navigate.md:16]:
```suggestion
All navigation-related configurations and navigators are neatly organized within the `src/navigators` folder. Here's a brief overview:
```

File: documentation/docs/04-Guides/04-Theming/02-Configuration.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
+---
+slug: /theming/configuration
+sidebar_label: configuration
```
Comment (jgautrais) [inline documentation/docs/04-Guides/04-Theming/02-Configuration.md:3]:
```suggestion
sidebar_label: Configuration
```

**Review by vbriand — COMMENTED** [review]
Very nice boilerplate! I've used this V4 for a few days and here are some issues I noticed.

**Comment by JeremyDolle** [comment]
Thanks a lot for the feedbacks and all the suggestions!

**Comment by vbriand** [comment]
You are welcome!

I have another quick suggestion to make though. While browsing through the React Navigation docs, I just noticed that they [recommend against wrapping the whole app with `SafeAreaView`](https://reactnavigation.org/docs/handling-safe-area/) to allow more control per screen, so I guess a small refactor to take that into account could be a good idea.

**Review by vbriand — COMMENTED** [review]
Not a huge issue but another thing I forgot to mention is that you left the files for the previous splash screen in `ios/MyApp/SplashIcon.imageset`.

**Pull request merged**